### PR TITLE
prelude: introduce Abort.error + test improvements

### PR DESCRIPTION
**Pull request by fwbrasil**

### CODE REVIEW

File: kyo-prelude/shared/src/test/scala/kyo2/AbortTest.scala [hunk 1]
```diff
@@ -11,71 +11,71 @@ class AbortsTest extends Test:
     "pure" - {
         "handle" in {
             assert(
-                kyo2.Abort.run[Ex1](kyo2.Abort.get[Ex1](Right(1))).eval ==
+                Abort.run[Ex1](Abort.get[Ex1](Right(1))).eval ==
```
Comment (fwbrasil) [inline kyo-prelude/shared/src/test/scala/kyo2/AbortTest.scala:14]: I think automated refactoring had unnecessarily introduced the package name.

File: kyo-prelude/shared/src/main/scala/kyo2/Abort.scala [hunk 2]
```diff
@@ -17,11 +17,12 @@ object Abort:
...
 
-    inline def fail[E](inline value: E): Nothing < Abort[E] =
-        Effect.suspendMap[Any](erasedTag[E], Fail(value))(_ => ???)
+    inline def fail[E](inline value: E): Nothing < Abort[E] = error(Fail(value))
 
-    inline def panic[E](ex: Throwable): Nothing < Abort[E] =
-        Effect.suspendMap[Any](erasedTag[E], Panic(ex))(_ => ???)
+    inline def panic[E](ex: Throwable): Nothing < Abort[E] = error(Panic(ex))
+
+    inline def error[E](e: Error[E]): Nothing < Abort[E] =
```
Comment (hearnadam) [inline kyo-prelude/shared/src/main/scala/kyo2/Abort.scala:24]: Shouldn't `e` be inlined here?

Do we want to make this package private? It should be stable, but I prefer to keep the public API small.

**Review by hearnadam — APPROVED** [review]

File: kyo-prelude/shared/src/main/scala/kyo2/Abort.scala (hunk 2 above)
Comment (fwbrasil) [reply kyo-prelude/shared/src/main/scala/kyo2/Abort.scala:24]: good catch! I've added the missing inline in `panic` as well

Comment (fwbrasil) [reply kyo-prelude/shared/src/main/scala/kyo2/Abort.scala]: > Do we want to make this package private? It should be stable, but I prefer to keep the public API small.

Makes sense, I've made it `private[kyo]`.

**Review by hearnadam — APPROVED** [review]

**Pull request merged**